packages/headless/src/controllers/commerce/facets/core/headless-core-facet.test.ts

```diff
@@ -0,0 +1,184 @@
+import {facetId} from '../../../core/facets/_common/facet-option-definitions';
+import {buildCoreFacet, Facet, FacetOptions} from './headless-core-facet';
+import {CommerceAppState} from '../../../../state/commerce-app-state';
+import {buildMockCommerceEngine, MockCommerceEngine} from '../../../../test';
+import {CommerceFacetRequest} from '../../../../features/commerce/facets/facet-set/interfaces/request';
+import {buildMockCommerceFacetSlice} from '../../../../test/mock-commerce-facet-slice';
+import {buildMockCommerceFacetRequest} from '../../../../test/mock-commerce-facet-request';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {buildMockFacetValue} from '../../../../test/mock-facet-value';
+import {fetchProductListing} from '../../../../features/commerce/product-listing/product-listing-actions';
+import {logFacetDeselect} from '../../../../features/facets/facet-set/facet-set-product-listing-v2-analytics-actions';
+
+
+describe('CoreCommerceFacet', () => {
```

Comment:
It seems like these are the tests for the headless-product-listing-facet controller, because the core controller doesn't dispatch analytics actions.

Reply:
Done in https://github.com/coveo/ui-kit/pull/3389